Show @lru_cache stats if PORTAGE_SHOW_LRU_CACHE_INFO env var is set

Flowdalic opened the pull request:
This introduces the PORTAGE_SHOW_LRU_CACHE_INFO environment variable.

Setting the PORTAGE_SHOW_LRU_CACHE_INFO environment variable allows to analyze the effectiveness and potential bottlenecks of the @lru_cache annotation used by portage. The cache information will be printed after dependency calculation and after emerge "finished".

Example output:
```
Portage @lru_cache information
use_reduce_cached: hit ratio: 71.25% (total: 48575, hits: 34611, misses: 13964) util: 100.00% (1024 / 1024)
encoded_length: hit ratio: 0.00% (total: 0, hits: 0, misses: 0) util: 0.00% (1024 / 0) 
vercmp: hit ratio: 69.50% (total: 1777, hits: 1235, misses: 542) util: 52.93% (1024 / 542) 
catpkgsplit: hit ratio: 66.90% (total: 191511, hits: 128129, misses: 63382) util: 100.00% (10240 / 10240) 
get_eapi_attrs: hit ratio: 100.00% (total: 267206, hits: 267200, misses: 6) util: 18.75% (32 / 6)
```

zmedico commented on lib/portage/util/portage_lru_cache.py:
```diff
@@ -0,0 +1,42 @@
+
+def show_lru_cache_info():
+    if not os.environ.get("PORTAGE_SHOW_LRU_CACHE_INFO"):
+        return
```
Maybe make this a FEATURES setting instead of environment variable?

Flowdalic replied:
I considered that. But it feels more like a debug thing that can be enabled via an env var, than a user facing env var. That is, I feel like to should be more hidden from users.That said, I wouldn't oppose a FEATURE if there is consensus that it's the right thing to do.

Flowdalic replied:
> I considered that. But it feels more like a debug thing, that can be enabled via an env var, than a user facing FEATURE setting. I feel like to should be more hidden from users.That said, I wouldn't oppose a FEATURE if there is consensus that it's the right thing to do.

eli-schwartz replied:
I don't think this makes sense as a FEATURES. That implies that users are supposed to report whether it's set as part of e.g. attaching `emerge --info` to a bug report... it also implies that users are expected to set this in make.conf so it prints on every run, rather than running it as part of development profiling.

zmedico replied:
Okay, I agree that FEATURES is not a good fit for this.

thesamesam commented:
Can you add something into NEWS as well as the man page (noting that it may disappear in future, output is unstable, it's for development only)?

Flowdalic commented:
> Can you add something into NEWS as well as the man page (noting that it may disappear in future, output is unstable, it's for development only)?

Added NEWS item, man page section, and rebased on the lastest master.
